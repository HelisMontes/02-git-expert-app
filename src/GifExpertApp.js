import React, {useState}from 'react';

export const GifExpertApp = () => {
    const [categories, setcategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);
    const handelAad = () => {
        // setcategories( [...categories, 'HunterXHunter'])
        setcategories(cats => [...categories, 'Hunter X Hunter'])
    }
    return (
        <>
            <h2>GifExpertApp</h2> 
            <hr/>
            <button onClick={ handelAad }>Enviar</button>
            <ul>
                {
                    categories.map(category => <li key={ category }> { category } </li> )
                }
            </ul>
        </>
    );
}