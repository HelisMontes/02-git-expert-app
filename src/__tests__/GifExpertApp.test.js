import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el componente GifExpertApp', () => {
    test('Cargando el componente GifExpertApp', () => {
        const wrapper =  shallow(<GifExpertApp />)
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar una lista de categorias', () => {
        const categories = ['Naruto', 'Hinata']
        const wrapper =  shallow(<GifExpertApp defaultCategories={ categories }/>)
        expect( wrapper ).toMatchSnapshot();
        expect ( wrapper.find('GifGrid').length ).toBe(categories.length)
    })
    
})
