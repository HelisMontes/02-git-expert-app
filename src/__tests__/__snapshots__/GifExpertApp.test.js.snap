// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Cargando el componente GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    categories="Naruto"
    key="Naruto"
  />
  <GifGrid
    categories="Hinata"
    key="Hinata"
  />
</Fragment>
`;
